public/class-cf7-grid-layout-public.php

```diff
@@ -1212,15 +1211,14 @@ public function wpcf7_mail_components($components, $cf7form, $cf7mail) {
            case 'tab':
            case 'table':
            case 'both':
-             $regex = '/^'.preg_quote($tag['name']);
              // Search for $_FILES where name match a tabbed file field
-             $submitted_fields = preg_grep($regex.'$/', $uploaded_files);
```

Comment:
not sure why you deleted this line.  the first tab fields use the original field names, subsequent tabs use <field-name>_tab-1, <field-name>_tab-2 ...
for rows (in tables) again, the same logic, the first row uses the naked field name, and subsequent rows have their field names afixed with the row suffix: <field-name>_row-1, <field-name>_row-2...

for tables within tabs, the first first row in the first tab is the naked field name, then the next row is <field-name>_row-1 and so on, In the 2nd tab you have the first row as <field-name>_tab-1, and the subsequent row as <field-name>_tab-1_row-1, <field-name>_tab-1_row-2 and so on.

Hence we need to still search for the naked field name.

Reply:
In fact, processing of naked upload files is already done by the CF7 validate methods. ("validate" in "submission.php" and "wpcf7_file_validation_filter" in "file.php")
Thus, we only need to call validation on duplicated fields (tabs, rows) in order to fill the uploaded_files array with missing FILES.